### DISCUSSION TITLE
BIRT.SOUR.ASSO

### DISCUSSION THREAD

**Discussion by mariannevanharten** [FamilySearch/GEDCOM#244] [Q&A]
I have a document about the registering of a birth at the municipality.
This documents consists of a few people (the person who want to register - often the father, the person at the municipality, witnesses).
I am searching for the right way to store this into a Gedcom (version 7).
When I write underneath part in my Gedcom file and check my file using the Gedcom Validator, this program tells me 
Invalid tag structure. Tag 'ASSO' is not permitted under tag 'SOUR'.
What is the right way to store these ASSO?

Best regards
Marianne

0 @I1@ INDI
1 NAME John /Doe/
1 BIRT
2 DATE 1 JAN 1850
2 PLAC Amsterdam
2 SOUR @S1@
....
0 @S1@ SOUR
1 ASSO @S2@
2 ROLE WITN
1 ASSO @S3@
2 ROLE .....

**Comment by Norwegian-Sardines** [comment]
I would think that a Witness is of an event rather than a witness of a source.  

A document may indicate a witness was in attendance, and this can be noted in the source either in a `NOTE` or `DATA.TEXT` .

A witness of an event would be noted using the `EVENT.ASSO` and this information can be "sourced" by using a Source_Citation `ASSO.SOUR`

Example (bolded):
0 @i1@ INDI
1 NAME John /Doe/
1 BIRT
2 DATE 1 JAN 1850
2 PLAC Amsterdam
**2 ASSO @X999@
3 ROLE WITN
3 SOUR @S1@**
2 SOUR @s1@
